### ISSUE TITLE
[BUG] Discard action doesn't work with Spamtrap

### ISSUE THREAD

**Issue by henrikalves** [rspamd/rspamd#3944]
Hello,
we are having the same problem as discussed in #3926 that's why we decided to create a bug report.
Using reject instead of discard as the action works but before version 3.0 it worked with discard aswell so something changed.

Here is the error:
```
2021-10-18 16:43:57 #14329(rspamd_proxy) ; lua; spamtrap.lua:67: spamtrap found:
2021-10-18 16:43:57 #14329(rspamd_proxy) ; lua; spamtrap.lua:78: action found: discard
2021-10-18 16:43:57 #14329(rspamd_proxy) ; proxy; lua_metric_symbol_callback: call to (SPAMTRAP_CHECK) failed (2): /usr/share/rspamd/plugins/spamtrap.lua:79: unknown action (null); trace: [1]:{[C]:-1 - set_pre_result [C]}; [2]:{/usr/share/rspamd/plugins/spamtrap.lua:79 - do_action [Lua]}; [3]:{/usr/share/rspamd/plugins/spamtrap.lua:136 - [Lua]};
```

the action has been defined so thats not the issue:
``` 
# /etc/rspamd/local.d/actions.conf

discard = {
   flags = ["no_threshold"],
   discard_on_reject = true
}
```
```
# /etc/rspamd/local.d/spamtrap.conf 

action = "discard";

map = file://$LOCAL_CONFDIR/local.d/maps.d/spamtrap.map;
learn_spam = true;
enabled = true;
```

**Comment by vstakhov** [comment]
It should be fixed by #3959. As a workaround you might define some large score threshold for `discard` action I suppose.

**Comment by gergernaut** [comment]
> It should be fixed by #3959. As a workaround you might define some large score threshold for `discard` action I suppose.

Can confirm the workaround, it works by setting e.g. `discard = 999.0;` in the local.d/actions.conf
Thanks a lot!

**Comment by leahoswald** [comment]
Looks like it isn't fixed by #3959 or it's a regression. Just observed with 3.8.4:

```
call to (SPAMTRAP_CHECK) failed (2): /usr/share/rspamd/plugins/spamtrap.lua:80: unknown action discard; trace: [1]:{[C]:-1 - set_pre_result [C]}; [2]:{/usr/share/rspamd/plugins/spamtrap.lua:80 - fun [Lua]}; [3]:{/usr/share/rspamd/lualib/fun.lua:34 - call_if_not_empty [Lua]}; [4]:{/usr/share/rspamd/lualib/fun.lua:672 - any [Lua]}; [5]:{/usr/share/rspamd/plugins/spamtrap.lua:149 - [Lua]};
```

The following doesn't work either.
```
discard = {
   flags = ["no_threshold"],
}
```

But the workaround still works.

**Comment by Habetdin** [comment]
I've met same issue with 3.7.5 but for `action = "quarantine"`:
```
#47(normal) <a6a581>; task; lua_metric_symbol_callback: call to (SPAMTRAP_CHECK) failed (2): /usr/share/rspamd/plugins/spamtrap.lua:80: unknown action quarantine; trace: [1]:{[C]:-1 - set_pre_result [C]}; [2]:{/usr/share/rspamd/plugins/spamtrap.lua:80 - fun [Lua]}; [3]:{/usr/share/rspamd/lualib/fun.lua:34 - call_if_not_empty [Lua]}; [4]:{/usr/share/rspamd/lualib/fun.lua:672 - any [Lua]}; [5]:{/usr/share/rspamd/plugins/spamtrap.lua:149 - <unknown> [Lua]};
```

---

> It should be fixed by #3959. As a workaround you might define some large score threshold for `discard` action I suppose.

@vstakhov The workaround worked, but it shouldn't be required, isn't it?
